src/index.js

```diff
@@ -309,10 +307,10 @@ function graphqlHTTP(options: Options): Middleware {
           operationName, result
         });
         response.setHeader('Content-Type', 'text/html; charset=utf-8');
-        sendResponse(response, payload);
+        response.end(payload);
       } else {
-        // Otherwise, present JSON directly.
-        const payload = JSON.stringify(result, null, pretty ? 2 : 0);
+        // Server will stringify our response object, we can return it directly
+        const payload = result || '';
```

Comment:
I agree that it's not ideal but I didn't want to change the existing logic with my PR as my purpose is just to bring compatibility with Restify.

The reason for the empty string is that at this level `result` can be `null` which is due to this  piece of code present 2 times, both at line 207 and 241
```
if (showGraphiQL) {
  return null;
}
```

If you don't want to alter the logic behind these 2 snippets then I have to deal with a possible `null` `result`.
The way your current code is dealing with that is to pass the `null` `result` to JSON.stringify like so `JSON.stringify(result, null, pretty ? 2 : 0)` which translates to `JSON.stringify(null, null, pretty ? 2 : 0)`; the result of this 'stringification' (which is an empty string) is then passed to `sendResponse`.
This means that I'm not introducing absolutely anything new by explicitly sending an empty string to `sendResponse` as this is exactly what was happening already, just it wasn't explicit and possibly not thought about/dealt with.